### ISSUE TITLE
Can not find the account, how long to recover the account?

### ISSUE THREAD

**Issue by loremwalker** [HelloZeroNet/ZeroNet#1434]
haven't user.json, username is token, how long to recover the account?

**Comment by loremwalker** [comment]
If  do not collect your account number, there will be a large number of zombie accounts in non-English international countries.

**Comment by Thunder33345** [comment]
sorry, you cant
unless you use another provider or something else

**Comment by loremwalker** [comment]
Permanent user name will not be logged out? This is not a good thing for me

**Comment by Thunder33345** [comment]
well the account is "claimed"
once it's claimed it will be owned by your forever
you can logout but you still own it
yes dead usernames may become a problem, but this applies on every social media anyways
there isnt a good solution, no one want their acount gone when they return after some time only for their account to be reclaimed which is not possible in zeronet because once something is signed it will stay signed

**Comment by Thunder33345** [comment]
if you didnt save the "secret" you will lost the keys to it,
and there's no way to proof you are the owner who lost the key
if there's a way to recover it can pose a huge security risk
just think of how recovery by email or phone number are always flawed

**Comment by loremwalker** [comment]
I hope to increase mailbox verification and phone verification，but it seems that it does not meet the essence of zeronet；
have a rip off, I hope to add some tips for using the interface

**Comment by Thunder33345** [comment]
zeronet is meant to be totally irrelevant to clearnet
if i need phone or mail to register, it can be violating my privacy
if i have optional phone or mail to register it pose a security risk, if someone hacked my email/phone now they have access to my zeroid

**Comment by krixano** [comment]
Besides, it's impossible because you'd need the private key which is only stored on one computer, the computer that had the account created. If you lose that private key, then you can't ever get it back.

**Comment by loremwalker** [comment]
thks very much, answered my question.

I think the account should be written on the home page，at least no one will make mistakes like me

**Comment by Thunder33345** [comment]
yes, we really need more notice to back your data up or you might loose it

**Comment by loremwalker** [comment]
i hope to increase support for special characters for registered accounts

**Comment by loremwalker** [comment]
such as `.`，`-`

**Comment by Thunder33345** [comment]
you can use other ids like kaffieID and cryptoID
which have no limit on what you can use

**Comment by loremwalker** [comment]
that's good,but how kaffie.bit download users.json?

**Comment by krixano** [comment]
When you register for an id, your local client registers it and puts it into your users.json file automatically. Everything is done locally, which is why kaffie.bit can't enforce username rules.

**Comment by loremwalker** [comment]
In other words, is kaffie.bit a temporary account?

**Comment by krixano** [comment]
No. I just answered this: `your local client registers it and puts it into your users.json file automatically`

**Comment by loremwalker** [comment]
In other words, the json data of kaffie.bit is inside the server, so it cannot be run locally. It needs to be browsed on the kaffie.bit platform. just kaffie.bit is zeronet proxy

**Comment by krixano** [comment]
Kaffie.bit it not a proxy. It is a zite. There is no server that kaffie.bit is on, it's local. All zites are downloaded locally. Basically, what the kaffie.bit zite does during username registration is call out to your client to create a private and public key for your new user and a certificate, and then adds that to your users.json file and associates it with kaffie.bit (basically by just putting the string "kaffie.bit").

KaffieId's are fully decentralized, unlike ZeroId, which is centralized.

I go over all of this in [the tutorial I linked previously](http://127.0.0.1:43110/DevCenter.bit/?/tutorials/users_databases_in_zeronet).

**Comment by loremwalker** [comment]
This seems really hard to understand

**Comment by loremwalker** [comment]
KaffieID still needs zeronetID support，too difficult

**Comment by krixano** [comment]
What do you mean zeronetID support?

**Comment by krixano** [comment]
That's because you saved that user on the proxy server, not on your client. The users.json file isn't magically synced up between all of your machines. Whatever computer you created that account on, that's where the data for that id is stored. In this case, you created the user on the proxy, so only the proxy has the user.

**Comment by loremwalker** [comment]
KaffieID can only use zeronet at this https://www.zerogate.tk，local can only register with zeroid but cannot use KaffieID，I originally wanted to register a local account with KaffieID

**Comment by imachug** [comment]
Just visit [id.kaffie.bit](http://127.0.0.1:43110/id.kaffie.bit) and register there on your local machine.

**Comment by krixano** [comment]
Go to KaffieId zite here: [http://127.0.0.1:43110/id.kaffie.bit/](http://127.0.0.1:43110/id.kaffie.bit/)

**Comment by Thunder33345** [comment]
You also need a hub that supports it
for example kaffie hub supports kaffie ID http://127.0.0.1:43110/hub.kaffie.bit/

**Comment by loremwalker** [comment]
I have not been able to express myself，In other words, registering KaffieID locally will write it into the users.json file.

**Comment by loremwalker** [comment]
I have registered KaffieID on different servers. The username is the same and I can register successfully

**Comment by loremwalker** [comment]
KaffieID is an independent existence, can exist the same username, there is no conflict

**Comment by Thunder33345** [comment]
Yes, username is just display name, as far as i know not related to the underlying signature
there can be 1000 people named bob@kaffieid.bit 
but each have individual files(someone please correct me if i am wrong)
so i afraid you cant move things from proxy to local

if you are using translator from chinese to english also feel free to not omit it, if needed i can try to translate it, as i am bilingual(know chinese and english)

**Comment by loremwalker** [comment]
在本地用KaffieID发布内容失败，但是操作是有效的；用zeroID发布内容却不会出现这类问题

**Comment by Thunder33345** [comment]
>在本地用KaffieID发布内容失败，但是操作是有效的；用zeroID发布内容却不会出现这类问题
using kaffieID on local peer broadcast failure, but it seems to work, using zeroID broadcast have problem as such

我试用了kaffieID在zerome但是如果我换了我的username整个网站都有问题
i tried using kaffieID on zerome, but if i change the username the whole website looks buggy when using it
我想需要site主人来修理
i think we need site developer to fix it

**Comment by loremwalker** [comment]
使用kaffieID账户注册用户名却无法搜索到，在本地中却可以查看相关的评论等操作，不知道其他的账户是否可见
![](https://s1.ax2x.com/2018/05/21/2FGaK.png)

**Comment by Thunder33345** [comment]
在这里我可以看到你的profile
http://127.0.0.1:43110/Me.ZeroNetwork.bit/?Profile/1SunAWK2VUT9GQK32MpwRfFPVgcBSJN9a/187QCanbYQFnSBjzGBR8foRuDFdzFUJ581/imloremwalker@zeroid.bit

**Comment by loremwalker** [comment]
我是指这个[kaffieID](http://127.0.0.1:43110/Me.ZeroNetwork.bit/?Profile/1SunAWK2VUT9GQK32MpwRfFPVgcBSJN9a/1EDKcHBp4en4MB61T4Awqb1TU2nknZkn8N/loremwalker@kaffie.bit)账户资料，它似乎好像不能搜索到任何相关消息，而且发出的信息，对方也基本没回复，但在本机上还是能看到发出的信息的

**Comment by Thunder33345** [comment]
>我是指这个kaffieID账户资料，它似乎好像不能搜索到任何相关消息，在新用户上也找不到kaffieID的账户，而且发出的信息，对方也基本没回复，但在本机上还是能看到发出的信息的
i am referring to this kaffieID account, it cant be seen anywhere

可以确认在这里不能看到那个帐号
can confirm i cant see it here,

可以说别的ID provider都有一些被支持的问题
can say other ID providers have problem begin support

**Comment by funny110** [comment]
丢了users.json，又没有备份，那你的帐号就无主了，任何人也找不回来了。

**Comment by User-Hrsat** [comment]
> 永久用户名不会被注销？这对我来说不是一件好事



> Permanent user name will not be logged out? This is not a good thing for me

Me too.😟

**Comment by imachug** [comment]
You can logout via sidebar. Many sites let you logout too.
